lib/fs/fs.go

```diff
@@ -406,9 +418,12 @@ type freeSpaceEntry struct {
 	freeSpace  uint64
 }
 
-// IsScheduledForRemoval returns true if the filename contains .must-remove. substring
-func IsScheduledForRemoval(filename string) bool {
-	return strings.Contains(filename, ".must-remove.")
+// IsScheduledForRemoval returns true if the supplied path (or name) is marked
+// for asynchronous removal. A directory is considered scheduled if
+// it contains a ".delete" sentinel file created by MustRemoveDirAtomic.
+func IsScheduledForRemoval(path string) bool {
+	sentinelPath := filepath.Join(path, ".delete")
```

Comment:
I suggest to add a fallback for previous logic - `strings.Contains(filename, ".must-remove.")`. It should cover version upgrade compatibility.